Bump json4s 3.2.11 to 3.3.0.RC1

seratch opened the pull request:
json4s 3.3.0.RC1 is out. https://github.com/json4s/json4s/releases/tag/3.3.0.RC1

This pull request bumps json4s version 3.2 to 3.3.0.RC1.

seratch commented:
TODO: atmosphere 2.2 upgrade #487

seratch commented:
Unfortunately Travis builds failed. scalatra-swagger and json4s 3.3.0.RC1 looks incompatible. Any thoughts?

rossabaker commented:
```
/home/travis/build/scalatra/scalatra/swagger/src/main/scala/org/scalatra/swagger/reflect/ScalaSigReader.scala:4: object tools is not a member of package scala
```

I haven't had time to try this hands on, but did we lose a transitive dependency that now needs to be explicit?

seratch commented:
Fixed. The issue is due to scalatra-swagger's scalap dependency. 

json4s embeds its own scalap instead of scala-lang's scalap dependency since 3.3.0 thanks to this PR https://github.com/json4s/json4s/pull/180

I just fixed to use json4s's one instead.
